brayns/parameters/ParametersManager.cpp

```diff
@@ -50,44 +48,59 @@ unsigned int levenshtein_distance(const std::string& s1, const std::string& s2)
 std::vector<std::string> findSimilarOptions(
     const std::string& name, const std::vector<std::string>& options)
 {
-    size_t bestDist = UINT_MAX;
-    std::vector<std::string> bestOptions;
     constexpr size_t MAX_SUGGESTIONS = 7;
 
-    for (const auto& optionName : options)
+    std::vector<std::string> subStringOptions;
+    std::vector<std::string> levenshteinOptions;
+
+    // Collect substring options
     {
-        const auto dist = levenshtein_distance(name, optionName);
-        if (dist < bestDist)
+        // Strip dashes
+        auto nameStrip = name;
+        nameStrip.erase(std::remove(nameStrip.begin(), nameStrip.end(), '-'),
+                        nameStrip.end());
+
+        // Suggest options containing the substring
+        for (const auto& optionName : options)
         {
-            bestOptions.clear();
-            bestDist = dist;
-        }
+            if (subStringOptions.size() >= MAX_SUGGESTIONS)
+                break;
 
-        if (dist == bestDist && bestOptions.size() < MAX_SUGGESTIONS)
-            bestOptions.push_back(optionName);
+            if (optionName.find(nameStrip) != std::string::npos)
+                subStringOptions.push_back(optionName);
+        }
     }
 
-    // Strip dashes
-    auto nameStrip = name;
-    nameStrip.erase(std::remove(nameStrip.begin(), nameStrip.end(), '-'),
-                    nameStrip.end());
-
-    // Also suggest options containing the substring
-    for (const auto& optionName : options)
+    // Collect best levenshtein distance options
     {
-        if (bestOptions.size() >= MAX_SUGGESTIONS)
-            break;
-
-        // Skip already present suggestions
-        if (std::find(bestOptions.begin(), bestOptions.end(), optionName) !=
-            bestOptions.end())
-            continue;
+        size_t bestDist = UINT_MAX;
 
-        if (optionName.find(nameStrip) != std::string::npos)
-            bestOptions.push_back(optionName);
+        for (const auto& optionName : options)
+        {
+            if (levenshteinOptions.size() >= MAX_SUGGESTIONS)
+                break;
+
+            const auto dist = levenshtein_distance(name, optionName);
+            if (dist < bestDist)
+            {
+                levenshteinOptions.clear();
+                bestDist = dist;
+            }
+
+            if (dist == bestDist &&
+                std::find(subStringOptions.begin(), subStringOptions.end(),
+                          optionName) == subStringOptions.end())
+                levenshteinOptions.push_back(optionName);
+        }
     }
 
-    return bestOptions;
+    // Merge suggestions giving precedence to substrings
+    auto output = subStringOptions;
+    output.insert(std::end(output), std::begin(levenshteinOptions),
+                  std::end(levenshteinOptions));
```

Comment:
Just a comment, why do you use std::end(..) instead of calling foo.end()? I would understand it in a template function, but here it makes the code less readable. I'd even say the same for using insert.

Don't you find this more readable and shorter to type:

    for (const auto& option : levenshteinOptions)
        output.push_back(option)

Reply:
Yes, that is better. I would like to see a 'vector += other_vector' or 'vector.append(other_vector)' in the standard library.